clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1815,16 +1804,6 @@ class SyclKernelDecompMarker : public SyclKernelFieldHandler {
     return true;
   }
 
-  bool handleSyclHalfType(const CXXRecordDecl *, const CXXBaseSpecifier &,
```

Comment:
I believe this is all you need to remove to avoid treating `half `like a special type.